File: client-js/evomaster-client-js/src/instrumentation/Action.ts
export default class Action {

    private readonly index: number;

    /**
     * A list (possibly empty) of String values used in the action.
     * This info can be used for different kinds of taint analysis, eg
     * to check how such values are used in the SUT
     */
    private readonly inputVariables: Set<String>;

    constructor(index: number, inputVariables: Set<String>) {
        this.index = index;
        this.inputVariables = new Set<String>(inputVariables);
    }

    getIndex(): number {
        return this.index;
    }

    getInputVariables(): Set<String> {
        //TODO consider to use Immutable.js or equivalent
        return new Set<String>(this.inputVariables);
    }
}

File: client-dotnet/src/EvoMaster.Instrumentation/Action.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;

namespace EvoMaster.Instrumentation {
    [Serializable]
    public class Action {
        private readonly int _index;

        /**
         * A list (possibly empty) of String values used in the action.
         * This info can be used for different kinds of taint analysis, eg
         * to check how such values are used in the SUT
         */
        private ISet<string> _inputVariables;

        public Action(int index, IEnumerable<string> inputVariables) {
            _index = index;
            //JAVA version: this.inputVariables = Collections.unmodifiableSet(new HashSet<>(inputVariables));
            _inputVariables = ImmutableHashSet.CreateRange(inputVariables);
        }

        public int GetIndex() {
            return _index;
        }

        public ISet<string> GetInputVariables() {
            return _inputVariables;
        }
    }
}